### PR TITLE
Changed from hanning to hann following scipy's new convention.

### LINKED ISSUE #3116: Scipy v1.9.0 renames scipy.signal.hanning to scipy.signal.hann
### Problem

Automated tests that use scipy v1.9.0 fail because `scipy.signal.hanning` has been removed in favor of `scipy.signal.hann` as noted in the [changelog](https://docs.scipy.org/doc/scipy/release.1.9.0.html). It seems like this has been in the works since [v1.1.0](https://docs.scipy.org/doc/scipy/release.1.1.0.html?highlight=hann).

### Proposed solution

Move to embracing the `scipy` convention by using `hann` instead of `hanning` in the code base. 

Alternatively, just make this change when calling `scipy.signal.get_window`. Note that `numpy` hasn't made this change and still uses `numpy.hanning` see this [discussion](https://github.com/numpy/numpy/issues/17303).

### PR DESCRIPTION
**Pull request by arkottke**
### What does this PR do?

Fixes #3116 by changing *hanning* to *hann* following the scipy statnd.

### Why was it initiated?  Any relevant Issues?

See #3116. Scipy v1.9.0 removed hanning.

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the "build_docs" tag to this PR.
      Docs will be served at [docs.obspy.org/pr/{branch_name}](https://docs.obspy.org/pr/) (do not use master branch).
      Please post a link to the relevant piece of documentation.
- [ ] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the "test_network" tag to this PR.
- [x] All tests still pass.
- [ ] Any new features or fixed regressions are covered via new tests.
- [ ] Any new or changed features are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [ ] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [ ] Add the "ready for review" tag when you are ready for the PR to be reviewed.

### CODE REVIEW

**Comment by megies** [comment]
Replacing it in the call syntax for resample as default for `window` is a no-brainer and that doesn't break anything. But I agree, maybe it's not necessary to rename it in the Lancosz part? To not break things. I don't think it's necessary that it's the same name in both parts.

**Comment by arkottke** [comment]
@megies yeah, I thought that was a bit drastic of a change, but I was hoping for consistency and then I discovered the numpy hasn't changed this as well.

We might want to add an alias of *hanning* to *hann* in `resample(...)` and note the is name is depreciated.

**Comment by megies** [comment]
> We might want to add an alias of _hanning_ to _hann_ in `resample(...)` and note the is name is depreciated.

Not sure if that's even needed. It was the default, so I imagine most people weren't explicitely specifying it. Furthermore, I'd imagine scipy has been showing a deprecation warning.

**Comment by megies** [comment]
In the lancosz part, it doesn't even has to do with numpy, as far as I see it's strictly our own naming in there and I think there's no reason to change that part.

**Comment by arkottke** [comment]
My brain just likes consistency so referring to the window as *hann* in one part of the code and *hanning* in another felt wrong, which is why I made the modifications that I did. I rolled back the changes to the interpolation routines to keep using *hanning*.

**Comment by megies** [comment]
I totally agree, it would be good to have the same naming consistently. But in this case, I'd say stability and not breaking existing codes is more important
Thanks for rolling that part back  =)

**Comment by megies** [comment]
OK, so this is pretty much ready to merge.. but, I actually think we should have this change in the maintenance branch for upcoming 1.3.1. It does not break anything and will make sure that code that is using resample without specifying a different window will keep working.

So.. I would argue this should be rebased on maintenance_1.3.x. Let me know if you need help with that. I can do that too if needed.

**Comment by arkottke** [comment]
It would be great if you can take care of rebasing on maintenance_1.3.x as
I won't be able to get to it for about 10 days.

**Comment by megies** [comment]
> It would be great if you can take care of rebasing on maintenance_1.3.x as I won't be able to get to it for about 10 days.

Done. I'll also force push your master branch to resemble our main master after merging this. Ideally you want to make dedicated branches for PRs =)

**Review by megies — APPROVED** [review]

**Comment by megies** [comment]
Actually github is pretty smart with their permissions, it wouldn't let me force push to your branch after merging this PR, so you'll have to reset your master branch yourself eventually (or just ignore your fork's master branch like I usually do, it's not really needed for anything ;)

File: CHANGELOG.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@ maintenance_1.3.x
 Changes:
  - obspy.core:
    * Fix exclusion of magnitude 0 events in Catalog.filter (see #3105)
+   * Trace.resample: Changed default for `window` to "hanning" following a name
```
Comment (liamtoney) [inline CHANGELOG.txt:7]: "hanning" and "hann" are reversed here.

**Review by liamtoney — COMMENTED** [review]
FYI, I noticed a small mixup in the changelog.

File: CHANGELOG.txt [hunk 2]
```diff
@@ -4,6 +4,9 @@ maintenance_1.3.x
...
+     change in scipy, "hann" is not recognized anymore in newest scipy (see
```
Comment (liamtoney) [inline CHANGELOG.txt:8]: And here.

Comment (megies) [reply CHANGELOG.txt:8]: Fixed, thanks!

**Comment by megies** [comment]
Fixed, thanks @liamtoney !

**Pull request merged**